DataStructure1/com/metacube/ArrayList.java

```diff
@@ -0,0 +1,160 @@
+package com.metacube;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.NoSuchElementException;
+
+/**
+ * @author Vinod Patel
+ *
+ */
+public class ArrayList<T> {
+	
+	private T[] objectList ;
+	private int index;
+	private int currentSize = 10;
+	
+	//initialize ArrayList with simple Object 
+	public ArrayList(){
+		objectList =(T[]) new Object[currentSize];
+		index = 0;
+	}
+	
+	
+	//get element from the given index
+	public T get(int index){
+		if(index < this.index){
+			return objectList[index];
+		}else{
+			throw new ArrayIndexOutOfBoundsException();
+		}
+	}
+	
+	
+	//add a element to list
+	public void add(int givenIndex, T newObject){
+		
+		if(givenIndex < 0 || this.size() <= givenIndex ){
```

Comment:
You should also check the currectSize of arraylist with maxSize of list. If list already has 10 elements then also new element should not get added.